backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceEventsHandler.java

```diff
@@ -0,0 +1,202 @@
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.interfaces.Api;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.services.BacktraceHandlerThread;
+import backtraceio.library.services.BacktraceReportSender;
+
+public abstract class BacktraceEventsHandler extends Handler {
+
+    private final static transient String LOG_TAG = BacktraceEventsHandler.class.getSimpleName();
+
+    protected final BacktraceHandlerThread backtraceHandlerThread;
+
+    /**
+     * Number of events in the queue
+     */
+    public int getCount() {
+        return events.size();
+    }
+
+    /**
+     * Maximum number of events in store. If number of events in store hit the limit
+     * BacktraceMetrics instance will send data to Backtrace.
+     */
+    private int maxNumEvents = 350;
+
+    public void setMaxNumEvents(int maxNumEvents) {
+        this.maxNumEvents = maxNumEvents;
+    }
+
+    public int getMaxNumEvents() {
+        return this.maxNumEvents;
+    }
+
+    /**
+     * Time between retries if metrics submission fails
+     */
+    private final int timeBetweenRetriesMillis;
+
+    /**
+     * List of events in the event queue
+     */
+    ConcurrentLinkedDeque<Event> events = new ConcurrentLinkedDeque<Event>();
+
+    /**
+     * Number of retry attempts
+     */
+    protected int numRetries = 0;
+
+    /**
+     * Http client
+     */
+    private final Api api;
+
+    /**
+     * Submission url
+     */
+    private final String submissionUrl;
+
+    /**
+     * User provided custom attributes
+     */
+    protected final Map<String, Object> customAttributes;
+
+    /**
+     * The application context
+     */
+    protected final Context context;
+
+    /**
+     * @param context
+     * @param api
+     * @param backtraceHandlerThread
+     * @param urlPrefix
+     * @param settings
+     */
+
+    public BacktraceEventsHandler(Context context, Map<String, Object> customAttributes,
+                                  Api api, final BacktraceHandlerThread backtraceHandlerThread, String urlPrefix, BacktraceMetricsSettings settings)
+    {
+        // This should always have a nonnull looper because BacktraceHandlerThread starts in the
+        // constructor and getLooper blocks until the looper is ready if the thread is started
+        super(backtraceHandlerThread.getLooper());
+        if (backtraceHandlerThread.getLooper() == null) {
+            throw new NullPointerException("Expected nonnull looper, this should not happen");
+        }
+        this.context = context;
+        this.customAttributes = customAttributes;
+        this.backtraceHandlerThread = backtraceHandlerThread;
+        this.api = api;
+        this.submissionUrl = settings.getBaseUrl() + "/" + urlPrefix + "/submit?token=" + settings.getToken() + "&universe=" + settings.getUniverseName();
```

Comment:
Hmm, now I can override 90% of the url but not everything.

Why not make the CTOR easier (remove urlPrefix) and add a second CTOR where full URLs are provided (or add to settings object, if the users want to specify full urls, ie some intercepting proxy or events-test or ....)

Reply:
Hmm, I'm not sure at all if it makes to override more than base URL - why? Other overrides might require additional server work? Route to the controller on the backend shouldn't change. 

Anyway, I think providing a full URL might give us more flexibility.

Reply:
Do we need to give user ability to specify any fully-qualified URL they want? As @konraddysput mentioned I think we expect that endpoints/routes should NOT change.

Reply:
I will mark this as resolved @vlussenburg @konraddysput